domain/src/test/java/org/oppia/android/domain/question/QuestionAssessmentProgressControllerTest.kt

```diff
@@ -994,6 +1003,515 @@ class QuestionAssessmentProgressControllerTest {
     assertThat(updatedState.ephemeralState.state.interaction.solution.solutionIsRevealed).isTrue()
   }
 
+  @Test
+  fun testRevealedSolution_forWrongAnswer_returnScore2OutOf3() {
```

Comment:
Here & below: I suggest factoring high-level actions & their sanity checks into different helper methods to only keep the high-level details relevant to the behavior being tested in the method body. For example, with refactoring this test could instead be:

```kotlin
fun testRevealedSolution_forWrongAnswer_returnScore2OutOf3() {
  // Arrange
  setUpTestApplicationWithSeed(questionSeed = 0)
  subscribeToCurrentQuestionToAllowSessionToLoad()
  startTrainingSession(TEST_SKILL_ID_LIST_2)
  submitWrongAnswerForQuestion2() // NOTE TO JESSICA: this could also perform verification as a sanity check
  submitSolutionIsRevealed()
  submitCorrectAnswerForQuestion2()

  // Act
  val grade = getExpectedSuccessfulGrade()

  // Assert
  assertThat(grade.numerator).isEqualTo(2)
  assertThat(grade.denominator).isEqualTo(3)
}
```

Notice how this can help focus the test body just to the details that the reader needs to see & hide the boilerplate to make it work. This is an easier test to verify, and reduces a lot of the overall code that we need to maintain for all of the tests being added.

I suggest making this or a similar clean up for all new tests here.

Reply:
Refactored, thanks!